### ISSUE TITLE
Not implementation of "some layers". No conforming implementation was found i.e. requested layer computation precision and output precision types are ignored, using the fastest implementation.

### ISSUE THREAD

**Issue by YouSenRong** [NVIDIA/TensorRT#2407]
## Description
I am using TensorRT in C++ to accelerate the inference using fp16 config, and the precision and the corresponding output datatype of some sensitive layers are set as kFLOAT32.

 However, there is some warning as "No implementation of layer zero__8 + (Unnamed Layer* 13) [Shuffle] obeys the requested constraints in strict mode. No conforming implementation was found i.e. requested layer computation precision and output precision types are ignored, using the fastest implementation.".

"No implementation of layer Mul__9 obeys the requested constraints in strict mode. No conforming implementation was found i.e. requested layer computation precision and output precision types are ignored, using the fastest implementation."

I had read this issue [BERT fp16 accuracy problem](https://github.com/NVIDIA/TensorRT/issues/1196), but there seems not solve the problem.

Which may lead to the undesired result such as "nan". How could I solve this problem, or is there any way I can make up the implementation by myself? Any advices would be appreciated!

## Environment

**TensorRT Version**: 7.2.3.4
**NVIDIA GPU**: Tesla T4
**NVIDIA Driver Version**: 460.73.01
**CUDA Version**: 11.2
**CUDNN Version**: 8.1.1.33-1
**Operating System**: ubuntu 16.04
**Python Version (if applicable)**: 
**Tensorflow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if so, version)**:

**Comment by zerollzeng** [comment]
For example, the warning means that TRT does not have any FP16 tactic(let's suppose you specify the layer with FP16 precision) for that layer, so it will fall back to FP32.

**Comment by zerollzeng** [comment]
I would suggest to try the latest TRT release, this might be get fixed in recent versions.

**Comment by YouSenRong** [comment]
> For example, the warning means that TRT does not have any FP16 tactic(let's suppose you specify the layer with FP16 precision) for that layer, so it will fall back to FP32.

I use FP16 and StrictType config flag, and set some sensitive layer to FP32 with setLayerPrecision(nvinfer1::DataType::kFLOAT32), so I guess the warning means that the kFLOAT32 has not implement, and use FP16 rather than fallback to FP32. I don't known if this guess is true. But the inference result is "nan" mean that some thing wrong, and it more likely happen on FP16 without fallback to FP32.

**Comment by YouSenRong** [comment]
> I would suggest to try the latest TRT release, this might be get fixed in recent versions.

Yes, I will try the latest version. Thanks a lot. If you have any new idea please let me know.

**Comment by YouSenRong** [comment]
Hello, @zerollzeng, @ttyio 
I have try the latest version 8.4.3. However, there are still some warnings as:
"[TRT] Weights [name=phase0_tf/task_9_gate_0/kernel:0] had the following issues when converted to FP16: - Subnormal FP16 values detected. "
Where, I have set all the layer with global precision i.e. kFLOAT.

Then, I set the warning layers with layer_precision to kFLOAT individually, the warning disappear, but the accuracy of the result is still degraded.

**Comment by YouSenRong** [comment]
> Can you reproduce this with [polygrahy](https://github.com/NVIDIA/TensorRT/tree/main/tools/Polygraphy)? the usage would be like `polygraphy run model.onnx --trt --onnxrt`, more details see https://github.com/NVIDIA/TensorRT/tree/main/tools/Polygraphy/examples/cli/run/01_comparing_frameworks
> 
> If the output is mismatch in FP32, please share your model with us for further investigation, might be an accuracy bug.

OK, let me have a try later, but sorry for that I can't share the model. I can share some information such as logs etc.

**Comment by YouSenRong** [comment]
@zerollzeng 
Actually, I can only set most of layers to FP32 in FP16 mode. For a little layers, i.e. kCONSTANT layers following with a unnamed kSHUFFLE layer, if I set these layer to FP32, I would get the error. 

For example, a part of the network as
![image](https://user-images.githubusercontent.com/21186538/198174912-e642e89d-718e-4ee7-8865-327c2d4d2ec9.png)

if I set the layer "phase0_tf/task_9_fc_0/bias:0" to FP32 in FP16 mode, I would get the error message as follows:

"[W] [TRT] Skipping tactic 0x0000000000000000 due to Myelin error: Mismatched type for tensor (Unnamed Layer_ 670) [Shuffle]_output', f32 vs. expected type:f16."
"[E] [TRT] 10: [optimizer.cpp::computeCosts::3626] Error Code 10: Internal Error (Could not find any implementation for node {ForeignNode[(Unnamed Layer* 1) [Constant] + (Unnamed Layer* 2) [Shuffle]...phase0_tf/predict_node]}.)
terminate called after throwing an instance of 'std::runtime_error'"

I guess it is the ConstantShuffleFusion leading to this problem. I have tried to set the output type of "(Unnamed Layer* 670) [Shuffle]", but it doesn't make sense.
Do you have any idea for this problem? Thanks a lot!

**Comment by YouSenRong** [comment]
I reproduce the problem on trtexec with "./bin/trtexec --onnx=saved_model.onnx --fp16 --precisionConstraints=obey --layerPrecisions=*:fp32", and get the error message as follows

[W] [TRT] Skipping tactic 0x0000000000000000 due to Myelin error: Mismatched type for tensor (Unnamed Layer_ 2) [Shuffle]_output', f32 vs. expected type:f16.
[E] Error[10]: [optimizer.cpp::computeCosts::3626] Error Code 10: Internal Error (Could not find any implementation for node {ForeignNode[(Unnamed Layer* 1) [Constant] + (Unnamed Layer* 2) [Shuffle]...phase0_tf/predict_node]}.)
[E] Error[2]: [builder.cpp::buildSerializedNetwork::636] Error Code 2: Internal Error (Assertion engine != nullptr failed. )
[E] Engine could not be created from network
[E] Building engine failed
[E] Failed to create engine from model or file.
[E] Engine set up failed

**Comment by zerollzeng** [comment]
@zhenhuaw-me Looks like a myelin mixed precision issue here, we might have some restrictions when offloading to Myelin, do you have any suggestions? Thanks!

**Comment by zhenhuaw-me** [comment]
@YouSenRong Thank you for reporting this issue. Setup type contraints for all layers can sometimes results in issues like this. It seems the error occurs during FP16 profiling, could you please share more logs (e.g. run with `trtexec --verbose`)?

And, does it make sense for you to skip the type constraint for this Shuffle layer?

**Comment by YouSenRong** [comment]
Thanks for the reply. 
Here is the [verbose log](https://drive.google.com/file/d/1yTzuKTJXF4yhB2EXCwJJvhLLFJiMlkcS/view?usp=share_link).

For some constant layers of my network, I may need to set it in kFLOAT precision, otherwise some value is overflow if in FP16 mode and lead to undesired result finally.

**Comment by zhenhuaw-me** [comment]
Hmmm, it's insteresting that FP32 is not profiled. @YouSenRong Could you please narrow to a mini repro mode thus we can look into the details?

**Comment by YouSenRong** [comment]
Sure. Does "mini repro mode" mean trying on a smaller model and get the less log?

**Comment by zhenhuaw-me** [comment]
@YouSenRong Yes exactually. You can try if `polygraphy inspect capability --help` can help you do that.

**Comment by nvpohanh** [comment]
I would also try TRT 8.5...

**Comment by ttyio** [comment]
Closing since no activity for more than 3 weeks, please reopen if you still have question, thanks!
